### PR TITLE
fix block grid size

### PR DESCRIPTION
**Pull request by 54way**
image
image
image

### CODE REVIEW

File: Editor/RaymarchShader.cs [hunk 1]
```diff
@@ -275,8 +275,11 @@ struct v2f {
                     half3 directionGrid = directionWorld;
                     half3 invDirectionGrid = 1.0 / directionGrid;
 
-                    float iBlockGridBlocks = floor((gridSize + blockSize - 1) / blockSize);
-                    float3 blockGridSize = iBlockGridBlocks * blockSize;
+                    int3 iGridSize = int3(round(gridSize));
+                    int  iBlockSize = int(round(blockSize));
+                    int3 iBlockGridBlocks = (iGridSize + iBlockSize - 1) / iBlockSize;
+                    int3 iBlockGridSize = iBlockGridBlocks * iBlockSize;
+                    half3 blockGridSize = half3(iBlockGridSize);
```
Comment (julienkay) [inline Editor/RaymarchShader.cs]:
```suggestion
                    float3 iBlockGridBlocks = floor((gridSize + blockSize - 1) / blockSize);
                    float3 blockGridSize = iBlockGridBlocks * blockSize;
```

The issue I've introduced was that iBlockGridBlocks should be a vector, not a scalar. The suggestion should bring back the correct behavior while still using float math, therefore not throwing warnings in Unity.

**Comment by julienkay** [comment]
Hi :)

thanks a lot for this PR and for flagging the issue! It seems I've introduced a bug with some recent changes with the intent of getting rid of Unity's shader warnings regarding the usage of integer divides ("integer divides may be much slower, try using uints if possible.")

I'd like to keep using floats so I've proposed a little change that I think would be a good fix. If you don't mind accepting I'll merge this PR.

Regarding the changes to the volume size calculations: I'm not strictly against the changes, but they seem to produce similar/the same results as the original code. Were there any scenes that gave you issues?

**Review by julienkay — CHANGES_REQUESTED** [review]

**Comment by 54way** [comment]
Hi~ 
Sorry for replying late. Recently I've got sick for 6 days, but I'm feeling better now.
I don't mind keeping using floats and making a better changes.
About the colume size caculation, toycar scene and spheres scene are cut by small volume which size is not big enough because the centroid of the volume does not necessarily coincide with the origin of the object's coordinate system.
toycar
spheres

**Comment by julienkay** [comment]
>About the colume size caculation, toycar scene and spheres scene are cut by small volume which size is not big enough because the centroid of the volume does not necessarily coincide with the origin of the object's coordinate system.

I see, you're completely right. Thanks for this contribution! Let me reopen the PR and merge this.

**Pull request merged**